src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/MicrosoftApiDesignGuidelinesAnalyzersResources.resx

```diff
@@ -514,7 +514,7 @@
     <value>To broaden the usability of a collection, implement one of the generic collection interfaces. Then the collection can be used to populate generic collection types.</value>
   </data>
   <data name="CollectionsShouldImplementGenericInterfaceMessage" xml:space="preserve">
-    <value>Collections should implement generic interface</value>
+    <value>Collection '{0}' inherits '{1}' without implementing '{2}'. Publicly-visible collections should implement the generic version to broaden usability.</value>
```

Comment:
FXCop has a different message, but this seem more clearer to me. This is the FXCop message for reference: http://index/?query=CollectionsShouldImplementGenericInterface&rightProject=DesignRules&file=CollectionsShouldImplementGenericInterface.cs&line=100

Should the message be: `Collection '{0}' directly or indirectly implements '{1}' without implementing '{2}'`?